src/plugins/query_enhancements/public/search/sql_search_interceptor.ts

```diff
@@ -62,6 +63,16 @@ export class SQLSearchInterceptor extends SearchInterceptor {
         .getDatasetService()
         .getType(datasetType);
       strategy = datasetTypeConfig?.getSearchOptions?.().strategy ?? strategy;
+
+      if (datasetTypeConfig?.languageOverrides?.SQL?.hideDatePicker === false) {
```

Comment:
Is this just for consistency? Because I don't see it add time range in the query automatically in sql search interceptor.

Reply:
the idea is that if there is a language override that sets `hideDatePicker: false`, then the search interceptor will not automatically insert the timeRange into query, but will pass `timeRange` to search strategy. Passing `timeRange` happens regardless of whether the search interceptor has logic to insert `timeRange`

Currently we don't have SQL query parser + builder so it's hard to insert time range to SQL, but if the search API for a specific dataset takes a SQL query and a separate time range parameter, then this will be useful